contracts/ResultManager.sol

```diff
@@ -14,39 +14,35 @@ contract ResultManager is AccessControlEnumerable {
         uint256 lastUpdatedTimestamp;
     }
 
-    bytes32 public constant RESULT_MANAGER_ADMIN_ROLE =
-        keccak256("RESULT_MANAGER_ADMIN_ROLE");
+    bytes32 public constant RESULT_MANAGER_ADMIN_ROLE = keccak256("RESULT_MANAGER_ADMIN_ROLE");
     bytes32 public constant FORWARDER_ROLE = keccak256("FORWARDER_ROLE");
 
     address public signerAddress;
 
-    /// @notice mapping for CollectionID -> Value Info
+    /// @notice mapping for Collection name -> Value Info
     mapping(bytes32 => Value) private _collectionResults;
 
-    event SignerUpdated(
-        address sender,
-        address indexed prevSigner,
-        address indexed newSigner
-    );
+    event SignerUpdated(address sender, address indexed prevSigner, address indexed newSigner);
 
     event ResultUpdated(Value value);
 
     error InvalidSignature();
     error InvalidMerkleProof();
     error ZeroResult();
+    error ZeroAddress();
 
     constructor(address _signerAddress) {
         _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        if (_signerAddress == address(0)) revert ZeroAddress();
         signerAddress = _signerAddress;
     }
 
     /**
      * @notice Updates the signer address
      */
-    function updateSignerAddress(
-        address _signerAddress
-    ) external onlyRole(RESULT_MANAGER_ADMIN_ROLE) {
+    function updateSignerAddress(address _signerAddress) external onlyRole(RESULT_MANAGER_ADMIN_ROLE) {
         emit SignerUpdated(msg.sender, signerAddress, _signerAddress);
+        if (_signerAddress == address(0)) revert ZeroAddress();
```

Comment:
@SamAg19 shouldn't we revert before emitting any event here? In case it is an invalid update

Reply:
fixed https://github.com/razor-network/bridge-proxy/pull/63/commits/d0e4a707dc5701acfbee69bb80ff7afc8659210e